enum describing SAM FLAG (for gui/ menu / etc... )

lindenb opened the pull request:
This pull request contains a new enum htsjdk.samtools.SAMFlag describing the available SAM flags. It can be used in a GUI, HTML form, menu... to display the available flags...

SAMFlag contains a static method returning the java.util.Set for a given integer

```java
public static Set<SAMFlag> getFlags(int flag)
```

I've added a small shortcut in src/java/htsjdk/samtools/SAMRecord.java
```java
   public final Set<SAMFlag> getSAMFlags() {
        return SAMFlag.getFlags( this.getFlags() );
    }
```

I've added a test under src/tests/java/htsjdk/samtools/SAMFlagTest.java

nh13 commented:
It seems like this duplicates some functionality in ```SAMRecord.java```.  Perhaps you could find a way to reconcile the two?

lbergelson commented on src/java/htsjdk/samtools/SAMFlag.java:
```diff
@@ -0,0 +1,92 @@
+ * SAM flags as enum, to be used in GUI, menu, etc...
+ */
+public enum SAMFlag {
+    READ_PAIRED(0x1),
```
Should these be renamed to reflect the more recent nomenclature? 

I.E.  `READ_PAIRED` ->  `MULTIPLE_READS_IN_TEMPLATE`

droazen replied:
Not sure that htsjdk is ready for the more modern terminology, since the newer names are inconsistent with the rest of the codebase. Couldn't this lead to confusion/error?

lindenb commented:
@nh13  I've removed the `final int` declarations in SAmRecord and I now use the values from SAMFLag.
@lbergelson I didn't know the new spec. I renamed the whole enum and added a human description for each item.

jacarey commented on src/java/htsjdk/samtools/SAMFlag.java:
```diff
@@ -0,0 +1,100 @@
+
+    /** @return the SAMFlag for the value 'flg' or null if it was not found */
+    public static SAMFlag valueOf(int flg) {
+        for (SAMFlag f : values()) {
```
flg -> flag

jacarey commented:
:+1:

nh13 commented:
@lindenb could you change the names of each enum value back to the previous names (ex. ```MULTIPLE_READS_IN_TEMPLATE``` -> ```READ_PAIRED_FLAG```.  We are not ready to modify all f the various methods in ```SAMRecord``` to the new terminology in the spec, given all the other code that relies on them.

lindenb commented:
@nh13 done.
